src/OrchardCore.Modules/OrchardCore.ContentTypes/Services/ContentDefinitionService.cs

```diff
@@ -167,13 +189,49 @@ await _contentDefinitionManager.AlterTypeDefinitionAsync(typeName, typeBuilder =
             })
         );
 
-        _contentDefinitionEventHandlers.Invoke((handler, context) => handler.ContentPartAttached(context), new ContentPartAttachedContext { ContentTypeName = typeName, ContentPartName = partName }, _logger);
+        var context = new ContentPartAttachedContext
+        {
+            ContentTypeName = typeName,
+            ContentPartName = partName,
+        };
+
+        _contentDefinitionEventHandlers.Invoke((handler, ctx) => handler.ContentPartAttached(ctx), context, _logger);
     }
 
     public async Task RemovePartFromTypeAsync(string partName, string typeName)
     {
+        var typeDefinition = await _contentDefinitionManager.LoadTypeDefinitionAsync(typeName);
+
+        if (typeDefinition == null)
+        {
+            return;
+        }
+
+        var partDefinition = typeDefinition.Parts.FirstOrDefault(p => string.Equals(p.Name, partName, StringComparison.OrdinalIgnoreCase));
+
+        if (partDefinition == null)
+        {
+            return;
+        }
+
+        var settings = partDefinition.GetSettings<ContentSettings>();
+
+        if (settings.IsCodeManaged)
+        {
+            _logger.LogError("The part '{PartName}' is code managed and cannot be removed from '{ContentType}' content type.", partName, typeName);
```

Comment:
I would not log here because the exception is part of the contract, some callers even hide it.

Reply:
done